src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -393,6 +393,14 @@ async Task IDurableOrchestrationClient.TerminateAsync(string instanceId, string
                 await this.client.TerminateInstanceAsync(state.OrchestrationInstance, reason);
 
                 this.traceHelper.FunctionTerminated(this.TaskHubName, state.Name, instanceId, reason);
+
+                this.traceHelper.OrchestratorStateChange(
+                    this.TaskHubName,
+                    state.Name,
+                    instanceId,
+                    FunctionType.Orchestrator,
+                    FunctionState.Terminated,
+                    isReplay: false);
             }
```

Comment:
Have we tested, on Azure, that terminating an orchestrator mid-execution won't lead it to accidentally emit a "Terminated" state-change log **and then** a "Completed" state-change log?

This was happening to us when when the following events took place:
1. orchestrator starts and schedules a 1 minute timer
2. before the 1 minute expires, the orchestrator is terminated. Therefore, the terminated state-change log is emitted
3. the time fires, and the orchestrator is resumed
4. the orchestrator performs a replay as a result of the timer firing. Therefore, it emits a completed state-change log.

I would appreciate if we could perform this test. It would help me gain confidence in the approach :)